### PR TITLE
Bf fix ratings

### PR DESCRIPTION
**Pull request by bf86**
Dive site rating now changes to reflect average review score. Also made some more styling changes and modified the test suite to reflect new button names.

### CODE REVIEW

File: app/models/dive_site.rb [hunk 1]
```diff
@@ -9,4 +9,16 @@ class DiveSite < ActiveRecord::Base
...
+      total_score = self.rating
+      self.reviews.each do |review|
+        total_score += review.score
+      end
+      total_score / (self.reviews.count + 1)
+    else
+      self.rating
+    end
+  end
 end
```
Comment (radavis) [inline app/models/dive_site.rb]: Redundant `hound_comment` detected.

Comment (radavis) [inline app/models/dive_site.rb]:
```
if reviews
  reviews.sum(:rating) / reviews.count.to_f
end
```

File: spec/features/user_views_details_of_a_dive_site_spec.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@
     expect(page).to have_content(dive_site.rating)
     expect(page).to have_content(dive_site.difficulty)
 
-    click_link dive_site.name
+    click_link "Details"
 
     expect(page).to have_content(dive_site.name)
     expect(page).to have_content(dive_site.location)
```
Comment (radavis) [inline spec/features/user_views_details_of_a_dive_site_spec.rb]: Create two or more reviews with different ratings and check that the page has the average of the scores.

**Comment by bf86** [comment]
Refactored average_rating method, added test, fixed hound violations.

**Pull request merged**